### PR TITLE
service-proxy: implement routing delegate header

### PR DESCRIPTION
**Pull request by rf**
r @Raynos @jcorbin 

If there's a routing delegate we defer to it as soon as possible.

Should this actually hit the rate limiting logic, at least for total RPS?

### CODE REVIEW

**Comment by Raynos** [comment]
Please also `handleRequest` for eager support.

File: service-proxy.js [hunk 1]
```diff
@@ -350,9 +354,10 @@ function handleLazily(conn, reqFrame) {
         }
     }
 
-    var serviceChannel = self.channel.subChannels[serviceName];
+    // use the rd (routing delegate) or the serviceName if there was no rd set
+    var serviceChannel = self.channel.subChannels[rd || serviceName];
```
Comment (Raynos) [inline service-proxy.js]: The weird thing is that we double count the rate limiter. we shouldnt increment rate limit for service if we do not route to service :(

**Comment by Raynos** [comment]
Weird that tests didnt fail without eager.

I think we default into lazy now and need to update the travis to re-run the entire test suite through eager.

Can you update the travis test config thingy ?

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ before_install:
 env:
 # Get some of them configs
 - HYPERBAHN_REMOTE_TEST_CONFIG=
-- HYPERBAHN_REMOTE_TEST_CONFIG=test/config/with_lazy_relay.json
+- HYPERBAHN_REMOTE_TEST_CONFIG=test/config/without_lazy_relay.json
```
Comment (jcorbin) [inline .travis.yml]: good catch, we did change the default recently, please break this out into a separate PR

File: service-proxy.js [hunk 3]
```diff
@@ -376,13 +385,15 @@ function handleRequest(req, buildRes) {
         return;
     }
 
-    if (self.isBlocked(req.headers && req.headers.cn, req.serviceName)) {
+    var rd = req.headers && req.headers.rd;
```
Comment (jcorbin) [inline service-proxy.js]: can we call this actually `routingDelegate` and then add a `var nextService = routingDelegate || req.serviceName;` and use it throughout?

**Comment by rf** [comment]
Split travis config fix into https://github.com/uber/hyperbahn/pull/180

**Comment by Raynos** [comment]
Good ship it.

**Comment by Raynos** [comment]
lgtm.
